airflow/providers/amazon/aws/operators/glue_session.py

```diff
@@ -0,0 +1,206 @@
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.hooks.glue_session import GlueSessionHook
+from airflow.providers.amazon.aws.triggers.glue_session import GlueSessionReadyTrigger
+from airflow.providers.amazon.aws.utils import validate_execute_complete_event
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class GlueCreateSessionOperator(BaseOperator):
+    """Create an AWS Glue Session.
+
+    AWS Glue is a serverless Spark ETL service for running Spark Jobs on the AWS
+    cloud. Language support: Python and Scala.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:GlueCreateSessionOperator`
+
+    :param session_id: session id
+    :param session_desc: session description
+    :param iam_role_name: AWS IAM Role for Glue Session Execution. If set `iam_role_arn` must equal None.
+    :param iam_role_arn: AWS IAM Role ARN for Glue Session Execution, If set `iam_role_name` must equal None.
+    :param num_of_dpus: Number of AWS Glue DPUs to allocate to this Session
+    :param create_session_kwargs: Extra arguments for Glue Session Creation
+    """
+
+    template_fields: Sequence[str] = (
+        "session_id",
+        "create_session_kwargs",
+        "iam_role_name",
+        "iam_role_arn",
+    )
+    template_ext: Sequence[str] = ()
+    template_fields_renderers = {
+        "create_session_kwargs": "json",
+    }
+    ui_color = "#ededed"
+
+    operator_extra_links = ()
+
+    def __init__(
+        self,
+        *,
+        session_id: str = "aws_glue_default_session",
+        session_desc: str = "AWS Glue Session with Airflow",
+        aws_conn_id: str = "aws_default",
+        region_name: str | None = None,
+        iam_role_name: str | None = None,
+        iam_role_arn: str | None = None,
+        num_of_dpus: int | None = None,
+        create_session_kwargs: dict | None = None,
+        wait_for_readiness: bool = True,
+        deferrable: bool = conf.getboolean("operators", "default_deferrable", fallback=False),
+        session_poll_interval: int | float = 6,
+        delete_session_on_kill: bool = False,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.session_id = session_id
+        self.session_desc = session_desc
+        self.aws_conn_id = aws_conn_id
+        self.region_name = region_name
+        self.iam_role_name = iam_role_name
+        self.iam_role_arn = iam_role_arn
+        self.num_of_dpus = num_of_dpus
+        self.create_session_kwargs = create_session_kwargs or {}
+        self.wait_for_readiness = wait_for_readiness
+        self.session_poll_interval = session_poll_interval
+        self.delete_session_on_kill = delete_session_on_kill
+        self.deferrable = deferrable
+
+    @cached_property
+    def glue_session_hook(self) -> GlueSessionHook:
```

Comment:
The common name for the hooks in AWS operators/sensors is `hook`
```suggestion
    def hook(self) -> GlueSessionHook:
```

If you use `AwsBaseOperator` this will define automatically